### DISCUSSION TITLE
Astro build integration

### DISCUSSION THREAD

**Discussion by alec-c4** [mantinedev/mantine#3171] [Feature requests]
Hi!
I'd like to ask you to create integration with https://astro.build I would be happy to help you with that.

**Comment by rtivital** [comment]
You can publish any package on npm that adds support for Astro. Currently, there are no plans to maintain such solution on our side.

**Comment by nfzv** [comment]
> Hi! I'd like to ask you to create integration with https://astro.build I would be happy to help you with that.

Hi! Are you using Astro with Mantine? Maybe you know how to get rid of "Invalid hook call" errors ?

**Comment by usamazansari** [comment]
Mantine with Astro would be super useful.

The only way I see out is applying `client:load` to the astro component which encapsulates `MantineProvider`. This defeats the purpose of the islands architecture of Astro - even if your pagesre server-rendered.

If there were a way we could run Mantine components or hooks without `MantineProvider` or have multiple instances of `MantineProvider` (any astro component using Mantine components should have a provider written in the astro component), the integration of Mantine with Astro would be the game changer!

**Reply by rtivital** [reply]
No, Astro does not support React context. From Astro [documentation](https://docs.astro.build/en/core-concepts/sharing-state/):

> UI frameworks like React or Vue may encourage “context” providers for other components to consume. But when partially hydrating components within Astro or Markdown, you can’t use these context wrappers.

All Mantine components depend on React context (MantineProvider component) and cannot be used with Astro. However you can still use all Mantine hooks from `@mantine/hooks` package in your Astro project.
